test/terra/backends/test_qasm_simulator_matrix_product_state.py

```diff
@@ -75,6 +77,8 @@ class TestQasmMatrixProductStateSimulator(
         QasmSnapshotStatevectorTests,
         QasmSnapshotProbabilitiesTests,
         QasmSnapshotStabilizerTests
+        #QasmSnapshotExpValPauliTests,
```

Comment:
What is the issue with the failing tests? Is it due to a bug in the expectation value method or something else?

Reply:
There are actually 3 issues in this PR:
1. bug in the interface of MPS expectation value, following the addition of SnapshotDataType. This was fixed by Yael.
2. bug in Pauli expectation value when `qubits` parameter is not given in ascending order.
3. bug in matrix expectation value when `qubits` parameter is not given in ascending order.
These were discovered because Yael added new tests after her change.
I am in the process of fixing 2 and 3, but would rather address them as separate issues (I will open these). The interface change (1) is more urgent, because that enables the users to use MPS in VQE. I stress that when the qubits are given in the default order, the expectation value works fine.